agent/lib/kontena/workers/service_pod_manager.rb

```diff
@@ -42,7 +42,7 @@ def on_update_notify(_, _)
 
     def populate_workers_from_master
       exclusive {
-        request = rpc_client.request("/node_service_pods/list", [node.id])
+        request = rpc_client.future.request("/node_service_pods/list", [node.id])
```

Comment:
Maybe, don't want to change these here.. tried to keep consumer code as close as possible to original.